### PR TITLE
Fixes 4925: fixup DBErrorToApi funcs

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

File: pkg/dao/snapshots.go [hunk 1]
```diff
@@ -219,17 +207,36 @@ func (sDao *snapshotDaoImpl) FetchModel(ctx context.Context, uuid string, includ
...
 }
 
+func SnapshotsDBToApiError(e error, uuid *string) *ce.DaoError {
+	if e == nil {
+		return nil
+	}
+
+	daoError := ce.DaoError{}
+	if errors.Is(e, gorm.ErrRecordNotFound) {
+		msg := "Task not found"
```
Comment (xbhouse) [inline pkg/dao/snapshots.go:222]:
```suggestion
		msg := "Snapshot not found"
```

File: pkg/dao/snapshots.go [hunk 2]
```diff
@@ -219,17 +207,36 @@ func (sDao *snapshotDaoImpl) FetchModel(ctx context.Context, uuid string, includ
...
+		if uuid != nil {
+			msg = fmt.Sprintf("Could not find snapshot with UUID %s", *uuid)
```
Comment (xbhouse) [inline pkg/dao/snapshots.go:224]: nitpicky but i think just to be consistent with the others, the error message here could be `Snapshot with UUID %s not found`

File: pkg/dao/repository_configs.go [hunk 3]
```diff
@@ -69,9 +69,21 @@ func DBErrorToApi(e error) *ce.DaoError {
...
+		if uuid != nil {
+			msg = fmt.Sprintf("Repository with UUID %s not found", *uuid)
+		}
+		daoErr = ce.DaoError{
+			Message:  msg,
+			NotFound: true,
+		}
+	} else {
+		daoErr = ce.DaoError{
+			Message:  "Database Error",
```
Comment (xbhouse) [inline pkg/dao/repository_configs.go:84]: i think this message should also display e.Error() if it's not a NotFound error

File: pkg/dao/templates.go [hunk 4]
```diff
@@ -21,7 +22,7 @@ type templateDaoImpl struct {
 	db *gorm.DB
 }
 
-func (t templateDaoImpl) DBToApiError(e error) *ce.DaoError {
+func (t templateDaoImpl) DBToApiError(e error, uuid *string) *ce.DaoError {
```
Comment (xbhouse) [inline pkg/dao/templates.go:25]: what do you think about either renaming this to TemplateDBToApiError and removing the receiver or also using a receiver on the other DBToApiError funcs and removing the object name? 

or using one DBToApiError method for all objects and just passing in the uuid and the object name? would that work?

File: pkg/dao/snapshots_test.go [hunk 5]
```diff
@@ -996,7 +996,7 @@ func (s *SnapshotsSuite) TestBulkDeleteNotFound() {
 
 	errs := sDao.BulkDelete(context.Background(), []string{s2.UUID, s3.UUID})
 	assert.Len(t, slices.DeleteFunc(errs, func(e error) bool { return e == nil }), 1)
-	assert.Equal(t, fmt.Sprintf("Could not find snapshot with UUID %s", s2.UUID), errs[0].Error())
+	assert.Equal(t, fmt.Sprintf("Could not find snapshot with UUID %s: record not found", s2.UUID), errs[0].Error())
```
Comment (swadeley) [inline pkg/dao/snapshots_test.go:999]: see comment about about "Could not find snapshot"

File: pkg/dao/templates.go (hunk 4 above)
Comment (jlsherrill) [reply pkg/dao/templates.go:25]: Some of them check for specific contraints, so as we are doin git today, i'm not sure we can do one method, but yes moving away from a reciever makes sense.

**Pull request merged**